getopt_long returns an int, not a char, and this matters a lot

https://github.com/fangfufu/httpdirfs/blob/d1a10d489c4b75d74ca5431c7641529a8742e85c/src/main.c#L172 contains a tragically common and subtle error; it should be `int`, not `char`, so that the comparison to `-1` in https://github.com/fangfufu/httpdirfs/blob/d1a10d489c4b75d74ca5431c7641529a8742e85c/src/main.c#L204-L206 works portably.  As written, it is broken on Arm for example, and the compiler will instead generate an infinite loop; see, for example, https://poudriere.cheribsd.org/hosts/freebsd-arm64/data/latest-per-pkg/fusefs-httpdirfs/1.2.3/cheribsd-aarch64-main.log, which says:

    src/main.c:206:39: warning: result of comparison of constant -1 with expression of type 'char' is always true [-Wtautological-constant-out-of-range-compare]
                            &long_index)) != -1) {

Unfortunately, this small error means that we always take the error path at https://github.com/fangfufu/httpdirfs/blob/d1a10d489c4b75d74ca5431c7641529a8742e85c/src/main.c#L76-L81 and, crucially, skip over the rest of the setup, both of the mountpoint and the network.  For maximum confusion, though, because we haven't actually done anything _wrong_ (because we haven't done anything at all, in some sense), the process ends up parked reading from `/dev/fuse` as it should, but without the mountpoint established, so it's listening for a message that's never coming.

--- a/src/fuse_local.c
+++ b/src/fuse_local.c
@@ -95,8 +95,8 @@ static int fs_open(const char *path, struct fuse_file_info *fi)
     if (!link) {
         return -ENOENT;
     }
-    if ((fi->flags & 3) != O_RDONLY) {
-        return -EACCES;
+    if ((fi->flags & O_RDWR) != O_RDONLY) {
+        return -EROFS;
     }
     if (CACHE_SYSTEM_INIT) {
         fi->fh = (uint64_t) Cache_open(path);

--- a/src/main.c
+++ b/src/main.c
@@ -169,7 +169,7 @@ void parse_config_file(char ***argv, int *argc)
 static int
 parse_arg_list(int argc, char **argv, char ***fuse_argv, int *fuse_argc)
 {
-    char c;
+    int c;
     int long_index = 0;
     const char *short_opts = "o:hVdfsp:u:P:";
     const struct option long_opts[] = {

